ruby_weather.rb

```diff
@@ -0,0 +1,98 @@
+require 'faraday'
+require 'json'
+require 'uri'
+
+#Get Postal-Code from User
+def get_postal_code(prompt)
+  puts prompt
+  postal_code = gets.chomp().to_s
+end
+
+#Validate the Postal-Code
+def validate_postal_code(postal_code, no_more_tries, count, limit, invalid,prompt, tries)
+  while postal_code.length != 6 and !no_more_tries
+    if count < limit
+      puts invalid
+      puts prompt
+      postal_code = gets.chomp().to_s
+      count += 1
+    else
+      no_more_tries = true
+    end
+  end
+
+  if no_more_tries
+    puts tries
+    return
+  end
+end
+
+#Get the Location Key
+def api_call_1(api_key, params, key_url)
+
+  res = Faraday.get(key_url) do |req|
+    req.params['apikey'] = api_key
+    req.params['q'] = params
+
+  end
+  resp = JSON.parse(res.body)
+end
+
+#Get the Current Conditions
+def api_call_2(api_key, params, key_url)
```

Comment:
Can we make this into a single `api_call` method without needing two separate ones?

Reply:
So, on the earlier version of the code, I set that to be just one api_Call for both actions but the code didn't work. I think the reason it didn't was that the 3rd params are different on both api’s

    res = Faraday.get(key_url) do |req|
    req.params['apikey'] = api_key
    req.params['q'] = params
______________________________________
    res = Faraday.get(key_url) do |req|
    req.params['apikey'] = api_key
    req.params['details'] = params

 Here if you notice both; one is referenced as 'q' and the other as 'details'. I tried using a common parameter for both requests but it doesn't work. any suggestions?

Reply:
You are hard coding the key for the parameter in each method (e.g. `req.params['details']). You could make the keys dynamic by assigning parameters slightly differently. 

```ruby
res = Faraday.get(key_url) do |req|
  req.options.params_encoder = Faraday::FlatParamsEncoder
  req.params = { roll: ['california', 'philadelphia'] } # See how this is different? We pass in a hash
end
```